imports/collections/schemas/catalog.js

```diff
@@ -1,20 +1,638 @@
+ * @property {Number} index required
+ * @property {Boolean} inventoryManagement required
+ * @property {Boolean} inventoryPolicy required
+ * @property {Boolean} isLowQuantity required
+ * @property {Boolean} isSoldOut required
+ * @property {Boolean} isTaxable required, default value: `false`
+ * @property {Number} length optional, default value: `0`
+ * @property {Number} lowInventoryWarningThreshold optional, default value: `0`
+ * @property {Metafield[]} metafields optional
+ * @property {Number} minOrderQuantity optional, default value: `1`
+ * @property {String} optionTitle optional
+ * @property {String} originCountry optional
+ * @property {CatalogPriceRange} price required
+ * @property {ProductPricingInfoByCurrency{}} pricing required
+ * @property {String} shopId required
+ * @property {String} sku optional
+ * @property {String} taxCode optional, default value: `"0000"`
+ * @property {String} taxDescription optional
+ * @property {String} title optional
+ * @property {Date} updatedAt optional
+ * @property {String} variantId required
+ * @property {Number} weight optional, default value: `0`
+ * @property {Number} width optional, default value: `0`
+ */
+export const CatalogProductVariant = new SimpleSchema({
+  "_id": {
+    type: String,
+    label: "Catalog product variant Id"
+  },
+  "barcode": {
+    type: String,
+    label: "Barcode",
+    optional: true
+  },
+  "createdAt": {
+    type: Date,
+    label: "Date/time this variant was created at",
+    optional: true
+  },
+  "height": {
+    type: Number,
+    label: "Height",
+    min: 0,
+    optional: true,
+    defaultValue: 0
+  },
+  "index": {
+    type: SimpleSchema.Integer,
+    label: "The position of this variant among other variants at the same level of the product-variant-option hierarchy"
+  },
+  "inventoryManagement": {
+    type: Boolean,
+    label: "True if inventory management is enabled for this variant"
+  },
+  "inventoryPolicy": {
+    type: Boolean,
+    label: "True if inventory policy is enabled for this variant"
+  },
+  "isLowQuantity": {
+    type: Boolean,
+    label: "Indicates that the product quantity is too low"
+  },
+  "isSoldOut": {
+    type: Boolean,
+    label: "Indicates when the product quantity is zero"
+  },
+  "isTaxable": {
+    type: Boolean,
+    label: "Indicates if a product is taxable",
+    index: 1,
+    defaultValue: false
+  },
+  "length": {
+    type: Number,
+    label: "Length",
+    min: 0,
+    optional: true,
+    defaultValue: 0
+  },
+  "lowInventoryWarningThreshold": {
+    type: SimpleSchema.Integer,
+    label: "Warn at",
+    min: 0,
+    optional: true,
+    defaultValue: 0
+  },
+  "metafields": {
+    type: Array,
+    label: "Metafields",
+    optional: true
+  },
+  "metafields.$": {
+    type: Metafield
+  },
+  "minOrderQuantity": {
+    type: SimpleSchema.Integer,
+    label: "Minimum quantity per item in an order",
+    defaultValue: 1,
+    optional: true
+  },
```

Comment:
Missing `options`? Because the type of the options items will be `CatalogProductVariant`, which is this object that we haven't yet defined, you'll have to restructure the code a bit. You could make a `VariantBaseSchema` with most of these and then clone and extend that to make the option schema and the variant schema, where only the variant schema would include the options array of option schema.